File: 10 - Python/Exercícios/04 - Vetores/Exercício 11.py
#Exercício - Dados Pessoas

N = int(input("Quantas pessoas serão digitadas? "))

altura = [0] * N
genero = [0] * N

soma = 0
homem = 0
mulher = 0

for i in range(N):

    altura[i] = float(input(f"Altura da {i+1}º pessoa: "))
    genero[i] = input(f"Gênero da {i+1}º pessoa: ")

    if genero[i] == "M" or genero[i] == "m":
        homem = homem + 1
    else:
        mulher = mulher + 1
        soma = soma + altura[i]

print("")

menor = altura[0]
maior = altura[0]

for i in range(1, N):

    if menor > altura[i]:
        menor = altura[i]

    if maior < altura[i]:
        maior = altura[i]

print("Menor altura =", menor)   
print("Maior altura =", maior)

media = soma / mulher

print("Média de altura das mulheres =", round(media,2))
print("Número de homens =", homem)

File: 07 - C++/Exercícios/04 - Vetores/Exercício 11.cpp
//Exercício - Dados Pessoas

#include <iostream>
#include <iomanip>

using namespace std;

int main(){

    int i, N, homem, mulher;
    double media, soma, menor, maior;

    cout << "Quantas pessoas serao digitadas? ";
    cin >> N;

    double altura[N];
    char genero[N];

    soma = 0;
    homem = 0;
    mulher = 0;

    for(i = 0; i < N; i++){

        cout << endl << "Altura da " << i+1 << "a pessoa: ";
        cin >> altura[i];
		cout << "Genero da " << i+1 << "a pessoa: ";
        cin >> genero[i];

        if(genero[i] == 'M' || genero[i] == 'm'){
            homem++;
        }
        else{
            mulher++;
            soma = soma + altura[i];
        }
    }

    media = (double) soma / mulher;

    menor = altura[0];
    maior = altura[0];

    for(i = 0; i < N; i++){

        if(menor > altura[i]){
            menor = altura[i];
        }

        if (maior < altura[i]){
            maior = altura[i];
        }

    }

    cout << fixed << setprecision(2);
    cout << endl << "Menor altura = " << menor << endl;
    cout << "Maior altura = " << maior << endl;
    cout << "Media das alturas das mulheres = " << media << endl;
    cout << "Numero de homens = " << homem;
    
    return 0;

}